server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -1437,11 +1460,28 @@ public ClusterState getClusterStateUsingDiff(ClusterMetadataManifest manifest, C
             indexRoutingTables.remove(indexName);
         }
 
-        return clusterStateBuilder.stateUUID(manifest.getStateUUID())
+        ClusterState clusterState = clusterStateBuilder.stateUUID(manifest.getStateUUID())
             .version(manifest.getStateVersion())
             .metadata(metadataBuilder)
             .routingTable(new RoutingTable(manifest.getRoutingTableVersion(), indexRoutingTables))
             .build();
+
+        if (checksumValidationEnabled && manifest.getClusterStateChecksum()!=null) {
+            validateClusterStateFromChecksum(manifest.getClusterStateChecksum(), clusterState);
+        }
+        return clusterState;
+    }
+
+    private void validateClusterStateFromChecksum(ClusterStateChecksum clusterStateChecksum, ClusterState clusterState) {
```

Comment:
What happens in case we throw IllegalStateException here?

Reply:
Publication for this node will fail and it will be kicked out of cluster, so it be easier to track if there's an issue during testing.